### PR TITLE
Pod-reloader update to the v0.0.124 and add reloadOnCreate option

### PR DESCRIPTION
**Pull request by Diaderius**
## Description
Update pod-reloader to the latest release (v0.0.124) and adding reloadOnCreate parameter.

## Why do we need it, and what problem does it solve?
Update due to new functionalities and features.
The `reloadOnCreate` parameter for workload redeployment after the related configMap/Secret is recreated.
Ref: #3091

## What is the expected result?
Once the secret or related ConfigMap/Secret is changed or removed and reapplied to the namespace, the workload is redeployed by default.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries
```changes
section: pod-reloader
type: feature
summary: Redeploy workload after ConfigMap/Secret recreation.
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Diaderius. I've started the workflow for label 'e2e/run/azure'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @Diaderius. I've started the workflow for label 'e2e/run/aws'...

**Review by yalosev — APPROVED** [review]

File: modules/465-pod-reloader/openapi/config-values.yaml [hunk 1]
```diff
@@ -39,3 +39,9 @@ properties:
       The same as in the pods' `spec.tolerations` parameter in Kubernetes;
 
       If the parameter is omitted or `false`, it will be determined [automatically](https://deckhouse.io/en/documentation/v1/#advanced-scheduling).
+  reloadOnCreate:
+    type: boolean
+    default: true
+    description: |
+      Option for workload redeployment after related config map/secret will be changed.
```
Comment (shvgn) [inline modules/465-pod-reloader/openapi/config-values.yaml:46]: Just a suggestion

```suggestion
      Rollout the workload when config map/secret changes by recreation.
```

File: modules/465-pod-reloader/openapi/doc-ru-config-values.yaml [hunk 2]
```diff
@@ -11,3 +11,7 @@ properties:
       Структура, аналогичная  `spec.tolerations` в Kubernetes Pod.
 
       Если ничего не указано или указано `false` — будет [использоваться автоматика](https://deckhouse.io/ru/documentation/v1/#выделение-узлов-под-определенный-вид-нагрузки).
+  reloadOnCreate:
+    type: boolean
+    description: |
+      Опция для перезапуска рабочей нагрузки в случае удаления и повторного применения измененного ресурса config map (cm) и secret. По умолчанию имеет значение 'true'.
```
Comment (z9r5) [inline modules/465-pod-reloader/openapi/doc-ru-config-values.yaml:17]:
```suggestion
      Разрешение перезапуска рабочей нагрузки в случае создания ConfigMap или Secret.
      
      Включение параметра будет приводить к перезапуску нагрузки не только при изменении соответствующего ConfigMap или Secret, но также и при его создании.
```

File: modules/465-pod-reloader/openapi/config-values.yaml [hunk 3]
```diff
@@ -39,3 +39,9 @@ properties:
       The same as in the pods' `spec.tolerations` parameter in Kubernetes;
 
       If the parameter is omitted or `false`, it will be determined [automatically](https://deckhouse.io/en/documentation/v1/#advanced-scheduling).
+  reloadOnCreate:
+    type: boolean
+    default: true
+    description: |
+      Rollout the workload when config map/secret changes by recreation.
```
Comment (z9r5) [inline modules/465-pod-reloader/openapi/config-values.yaml:46]:
```suggestion
      Rollout the workload in case of creating a ConfigMap or Secret.
```

**Review by z9r5 — APPROVED** [review]

**Pull request merged**